### ISSUE TITLE
Directory specific compiler flags

### ISSUE THREAD

**Issue by kresan** [R3BRootGroup/R3BRoot#176]
Compiler flags should be the same for all sub-folders and are to be defined on the level of top CMakeLists.txt. Please remove directory-specific definition in neuland/CMakeLists.txt.

Cheers,
Dima

**Comment by janmayer** [comment]
Note that when I add ` -Wall -Wextra -Wformat-security` to the topmost CMakeLists, I get a huge amount of warnings.

Some are meaningless, but imho all should get fixed. Could help to find or prevent bugs, and improve readability (I'm looking at you, `unused-parameter`! :dizzy_face:)

```bash
$ cat build.log | grep warning | grep -oE '\[(.*)\]' | sort | uniq -c
      2 [-Wcomment]
    123 [-Wextra]
    184 [-Wignored-qualifiers]
     57 [-Wmaybe-uninitialized]
      1 [-Wmisleading-indentation]
    147 [-Wreorder]
     63 [-Wsign-compare]
     48 [-Wunused-but-set-variable]
    222 [-Wunused-parameter]
```

```bash
$ cat build.log | grep warning | grep -oE '\/R3BRoot\/(.*)\/' | sort | uniq -c
     23 /R3BRoot/califa/
    109 /R3BRoot/califa/unpack/
     10 /R3BRoot/dch/
     16 /R3BRoot/dtof/
      7 /R3BRoot/evtvis/
     14 /R3BRoot/fi4/
      4 /R3BRoot/fi5/
      4 /R3BRoot/fi6/
     39 /R3BRoot/field/
      7 /R3BRoot/gfi/
     78 /R3BRoot/land/
     34 /R3BRoot/land/calibration/
     13 /R3BRoot/los/
      6 /R3BRoot/lumon/
      7 /R3BRoot/mfi/
      8 /R3BRoot/mtof/
     25 /R3BRoot/psp/
      6 /R3BRoot/psp/par/
     25 /R3BRoot/r3bbase/
      3 /R3BRoot/r3bdata/
      3 /R3BRoot/r3bdata/actarData/
      6 /R3BRoot/r3bdata/califaData/
      6 /R3BRoot/r3bdata/dchData/
      5 /R3BRoot/r3bdata/dtofData/
      1 /R3BRoot/r3bdata/fi4Data/
      3 /R3BRoot/r3bdata/gfiData/
     10 /R3BRoot/r3bdata/landData/
      1 /R3BRoot/r3bdata/lumonData/
      3 /R3BRoot/r3bdata/mfiData/
     11 /R3BRoot/r3bdata/mtofData/
      1 /R3BRoot/r3bdata/neulandData/
     11 /R3BRoot/r3bdata/pspData/
     22 /R3BRoot/r3bdata/startrackData/
     17 /R3BRoot/r3bdata/tofData/
     45 /R3BRoot/r3bdata/trackerData/
      3 /R3BRoot/r3bdata/vetoData/
     13 /R3BRoot/r3bdata/xballData/
     19 /R3BRoot/r3bgen/
      5 /R3BRoot/sfi/
     11 /R3BRoot/startrack/
     22 /R3BRoot/startrack/unpack/
     61 /R3BRoot/startrack/unpack/B4Ucesb/
     20 /R3BRoot/startrack/unpack/par/
      3 /R3BRoot/strawtubes/
     87 /R3BRoot/tof/
     59 /R3BRoot/tracker/
      1 /R3BRoot/tracking/
      9 /R3BRoot/xball/
```

**Comment by kresan** [comment]
I agree, the warnings you mentioned should be eliminated. Will open a new issue for that. The current one is fixed in #186
